### PR TITLE
Dependency fixes

### PR DESCRIPTION
**Pull request by jglick**
Fixes a linkage error when updating `workflow-support` to 2.0.

Critical, but this change is disruptive enough that we may want to hold off on it. I will in parallel work on an alternate compatibility fix.

@reviewbybees esp. @svanoort

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,8 +46,7 @@
     </scm>
 
     <properties>
-        <jenkins.version>1.609.3</jenkins.version>
-        <workflow.version>1.14</workflow.version>
+        <jenkins.version>1.642.3</jenkins.version>
```
Comment (svanoort) [inline pom.xml]: Why are we bumping to 1.642.3 jenkins dependency?  This removes usability for many customers.

File: ui/gulpfile.js [hunk 2]
```diff
@@ -46,3 +46,4 @@ builder.bundle('src/main/js/stageview.js')
 // Explicitly setting the task list so as to disable jshint
 // TODO: Remove the line below + fix jshint build errors
 builder.defineTasks(['test', 'bundle', 'rebundle']);
+builder.defineTask('lint', function() {}); // https://github.com/jenkinsci/plugin-pom/pull/20#issuecomment-206374472
```
Comment (svanoort) [inline ui/gulpfile.js]: Awesome.  I was building a PR that included this.

File: pom.xml (hunk 1 above)
Comment (amuniz) [inline pom.xml]: What was producing the linkage error? And why it needs to bump baseline to 1.642.3?

File: pom.xml [hunk 3]
```diff
@@ -28,7 +28,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>2.3</version>
+        <version>2.6</version>
```
Comment (svanoort) [inline pom.xml]: This I like.  I have a commit adding it in a branch I wanted to push.

File: pom.xml (hunk 1 above)
Comment (amuniz) [inline pom.xml]: Ok, I think the answer to my second question is because Pipeline 2.0 requires that baseline.

Comment (jglick) [inline pom.xml]: Correct.

**Comment by svanoort** [comment]
@jglick Can this be closed now that the other PR is merged?

**Comment by jglick** [comment]
No, this is still useful for the future, because the `optional` dependency is just a hack to allow the class loader linkage to be made. It does not, for example, block you from disabling `pipeline-input-step`, and the current `rest-api` code is not prepared for the possibility that this dependency is missing. So either we keep the policy of a hard dep on the presence of the `input` step, in which case something like this PR is needed as soon as we are comfortable increasing the core dependency; or the processing of `input` is made into a proper extension, either loaded optionally from `pipeline-rest-api`, or factored into a separate plugin in the style of `cloudbees-workflow-ui`.

**Comment by jglick** [comment]
> block you from disabling `pipeline-input-step`

Checked this in 1.642.3 and the UI _does_ block you from disabling it, despite the optional status of the dependency. Perhaps a core bug fixed later?

**Comment by svanoort** [comment]
@jglick The next stage view release will retain the 1.609.x dependency, then the one after it will take up a version of this along with dependency on 1.642.x in order to use the new FlowScanner algorthms.

**Comment by svanoort** [comment]
This is rolled up into https://github.com/jenkinsci/pipeline-stage-view-plugin/pull/25 now.

**Pull request merged**